### PR TITLE
Fix an edge case when spending zero-blindingfactor inputs

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by sipa** [comment]
@TheBlueMatt can you report whether this works?

**Comment by TheBlueMatt** [comment]
It works for me in limited testing.

**Comment by sipa** [comment]
Significantly reworked the code. It will now always refuse to create a zero blinding factor, as those are not actually private (both because the value is automatically visible from the commitment, and because it means you could trace the transaction history and find a complete covering set of unblinded coins that provides all the value for this output).

The wallet code will automatically add a dummy output in this case. The RPCs will fail if this is attempted, and suggest adding a dummy output.

**Comment by jonasnick** [comment]
ACK

File: src/wallet.cpp [hunk 1]
```diff
@@ -1673,11 +1671,17 @@ bool CWallet::CreateTransaction(const vector<CSend>& vecSend, const vector<CTxIn
...
-                    return false;
+                if (!BlindOutputs(input_blinds, output_blinds, output_pubkeys, txNew)) {
+                    // We need an extra blinded output
+                    // TODO: if fBlindedOutputs, don't use an OP_RETURN but create an (extra) change output
+                    // instead, as this does not actually provide better privacy.
+                    CTxOut newTxOut(0, CScript() << OP_RETURN);
+                    txNew.vout.push_back(newTxOut);
+                    output_pubkeys.push_back(GetBlindingPubKey(newTxOut.scriptPubKey));
+                    // Now it has to succeed
+                    bool ret = BlindOutputs(input_blinds, output_blinds, output_pubkeys, txNew);
```
Comment (instagibbs) [inline src/wallet.cpp]: output_blinds needs to be expanded by 1 as well or it fails assert

Comment (sipa) [inline src/wallet.cpp]: Fixed.

**Comment by instagibbs** [comment]
Included in `elements-0.13.1`, closing

**Pull request closed without merging**